File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/annotations/OpenApiRequestBodyInstance.java
package net.dzikoysk.openapi.processor.annotations;

import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
import net.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper;

import javax.lang.model.element.AnnotationMirror;
import java.lang.annotation.Annotation;

public final class OpenApiRequestBodyInstance extends AnnotationMirrorMapper {

    public OpenApiRequestBodyInstance(AnnotationMirror mirror) {
        super(mirror);
    }

    public OpenApiContentInstance[] content() {
        return getArray("content", AnnotationMirror.class, OpenApiContentInstance::new, OpenApiContentInstance[]::new);
    }

    public String description() {
        return getString("description");
    }

    public boolean required() {
        return getBoolean("required");
    }

    public Class<? extends Annotation> annotationType() {
        return OpenApiRequestBody.class;
    }

}

File: openapi-processor/src/main/kotlin/com/dzikoysk/openapi/processor/annotations/OpenApiRequestBodyInstance.kt
package com.dzikoysk.openapi.processor.annotations

import com.dzikoysk.openapi.annotations.OpenApiRequestBody
import com.dzikoysk.openapi.processor.processing.AnnotationMirrorMapper
import javax.lang.model.element.AnnotationMirror

internal class OpenApiRequestBodyInstance(mirror: AnnotationMirror) : AnnotationMirrorMapper(mirror) {

    fun content(): List<OpenApiContentInstance> =
        getArray("content", AnnotationMirror::class.java) { OpenApiContentInstance(it) }

    fun description(): String =
        getString("description")

    fun required(): Boolean =
        getBoolean("required")

    fun annotationType(): Class<out Annotation> =
        OpenApiRequestBody::class.java

}